### ISSUE TITLE
custom cookbook extract failure after downloading from S3

### ISSUE THREAD

**Issue by KangYoungOh** [aws/opsworks-cookbooks#214]
Apps deploy with opsworks fails since 5th, September.
1. custom cookbook zip file exists in S3.
2. cookbook is downloaded successfully.
   /tmp/opsworks....../archive.d/archive
3. cookbook extract fails ?
   site-cookbooks archive file is same with one in tmp folder
   /opt/aws/opsworks/current/site-cookbooks/archive
   (before 5th, September it's extracted successfully)
4. when deploy with opsworks, error occurs (custom cookbook not found)
   ERROR: Caught exception while compiling OpsWorks custom run list: Chef::Exceptions::CookbookNotFound - Cookbook ~~~~~~~~~~ not found. If you're loading `````````cookbook from another cookbook, make sure you configure the dependency in your metadata - /opt/aws/opsworks/releases/20140909103909_234/vendor/gems/chef-11.4.4/bin/../lib/chef/cookbook/cookbook_collection.rb:38:in`initialize'

**Comment by githuesch** [comment]
Can you check what /tmp/opsworks....../archive.d/archive actually looks like? Our guess is that it's an XML file with an error message instead of an actual archive of your cookbooks.

**Comment by KangYoungOh** [comment]
[root@openapi-1 archive.d]# pwd
/tmp/opsworks20140912-24516-1iglvoq/archive.d
[root@openapi-1 archive.d]# ll
total 40
-rw-r--r-- 1 root root 37754 Sep 12 09:26 archive
[root@openapi-1 archive.d]# file archive
archive: Microsoft OOXML

File size is same with the one in S3.
After local downloading the archive file, it's uncompressed successfully. 

I tested old cookbook(which was operated normally), but the same error occurs.

**Comment by KangYoungOh** [comment]
manual extract execution error
/bin/extract ~~~~~~~~/archive
unhandled archive format '/tmp/opsworks20140916-386-d5q057/archive.d/archive: Microsoft OOXML' - not extracting

manual unzip execution success
"unzip -d #{directory} #{archive}" is uncompressed successfully on server.

**Comment by githuesch** [comment]
The output of the `file` utility indicates that it's XML, as we assumed. I would be surprised if `unzip` was really able to extract anything useful. Once you have another downloaded file that's identified as XML, please look at the content instead of just passing it on to `unzip`. It should tell you that there was a permission problem or something similar.

**Comment by KangYoungOh** [comment]
Manual unzip correctly uncompress the cookbook. (file contents is ok too.) :)

**Comment by githuesch** [comment]
Then I don't think you passed the same file to `file` and `unzip`. If your file is a zip archive, `file` should print something like "Zip archive data, at least v2.0 to extract", not Microsoft OOXML.

Furthermore I'm not sure what your latest update actually means. Is your problem solved now?

**Comment by githuesch** [comment]
If unzipping your archive really works with `unzip` even though `file` reports the file to be an XML document, please tell us which tool you used to create that zip file. It would help us make the extraction more robust.

**Comment by KangYoungOh** [comment]
I used 7-zip on local windows.
then, opsworks unzip well until last month.

**Comment by KangYoungOh** [comment]
Thanks, I'll try other zip method on tomorrow.

**Comment by gfawcettpq** [comment]
We've been having the same issue for two months now as reported here https://forums.aws.amazon.com/thread.jspa?threadID=157289&tstart=0.  In our case it had affected war artifacts built with gradle.  It's caused by an improper regex in the version of the file utility that was released to Amazon Linux late July.  We ended up custom compiling the magic file used by file to remove the OOXML definition and have been deploying it to our OpsWorks instances as part of our setup run lists.

That being said, it looks like the fixed version of file is now in Amazon's repository so we can probably stop deploying our magic file.  File 5.19-4.19 is identifying our war artifacts as Zip files correctly
